app/unit-tests/src/org/commcare/android/tests/DataPullTaskTest.java

```diff
@@ -126,13 +126,14 @@ public void dataPullSelfIndexingCaseTest() {
     @Test
     public void asyncRestoreTest() {
         installAndUseLocalKeys();
+        int initialCount = LocalReferencePullResponseFactory.getNumRequestsMade();
         runDataPullWithAsyncRestore();
 
         Assert.assertEquals(DataPullTask.PullTaskResult.DOWNLOAD_SUCCESS, dataPullResult.data);
         
         // Indicates that the task executed all of the retries we indicated, and then successfully
         // parsed the final success response
-        Assert.assertEquals(4, LocalReferencePullResponseFactory.getNumRequestsMade());
+        Assert.assertEquals(4, LocalReferencePullResponseFactory.getNumRequestsMade() - initialCount);
```

Comment:
@shubham1g5 Ahhh, is this why this test fails sometimes??

Reply:
Yeah!

Reply:
amazing find, thank you